iceoryx_posh/test/moduletests/test_runtime_ipc_interface.cpp

```diff
@@ -0,0 +1,328 @@
+// Copyright (c) 2019 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2021 - 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_dust/posix_wrapper/message_queue.hpp"
+#include "iceoryx_dust/posix_wrapper/named_pipe.hpp"
+#include "iceoryx_hoofs/internal/posix_wrapper/unix_domain_socket.hpp"
+#include "iceoryx_hoofs/platform/platform_settings.hpp"
+#include "iceoryx_posh/internal/runtime/ipc_interface_base.hpp"
+
+#include "test.hpp"
+
+#include <chrono>
+
+namespace
+{
+using namespace ::testing;
+using namespace iox;
+using namespace iox::posix;
+using namespace iox::units::duration_literals;
+
+#if defined(__APPLE__)
+using IpcChannelTypes = Types<runtime::IpcInterface<UnixDomainSocket>>;
+#elif defined(_WIN32)
+using IpcChannelTypes = Types<runtime::IpcInterface<NamedPipe>>;
+#elif defined(unix) || defined(__unix) || defined(__unix__)
+using IpcChannelTypes = Types<runtime::IpcInterface<UnixDomainSocket>, runtime::IpcInterface<NamedPipe>>;
+#else
+using IpcChannelTypes = Types<runtime::IpcInterface<UnixDomainSocket>, runtime::IpcInterface<NamedPipe>>;
+#endif
+
+constexpr char goodName[] = "channel_test";
+constexpr char anotherGoodName[] = "horst";
+constexpr char slashName[] = "/miau";
+
+/// @req
+/// @brief This test suite verifies that the abstract interface IpcChannelType is fulfilled by both the UnixDomainSocket
+/// class and the MessageQueue class
+/// @pre server and client object are allocated and move to the member object of the fixture
+/// @post StdErr is capture and outputed to StdCout
+/// @note Specific functionality of the underlying implementations of an IpcChannelType are tested in
+/// "UnixDomainSocket_test"
+template <typename T>
+class IpcChannel_test : public Test
```

Comment:
Shouldn't the test be renamed?

```suggestion
class IpcInterface_test : public Test
```